apps/ewallet/lib/ewallet/gates/transaction_consumption_gate.ex

```diff
@@ -144,21 +144,55 @@ defmodule EWallet.TransactionConsumptionGate do
           |> transfer(request.type)
       end
     else
+      {:ok, consumption} -> {:ok, consumption}
       error when is_atom(error) -> {:error, error}
       error -> error
     end
   end
 
+  def get_with_idempotency_token(idempotency_token) do
+    {:ok, TransactionConsumption.get_by(idempotency_token: idempotency_token)}
+  end
+
   defp validate_max_consumptions_per_user(request, balance) do
-    case request.max_consumptions_per_user do
-      nil ->
-        {:ok, balance}
-
-      max ->
-        case TransactionConsumption.all_active_for_user(balance.user_uuid) do
-          [] -> {:ok, balance}
-          _ -> {:error, :max_consumptions_per_user_reached}
-        end
+    validate_max_consumptions_per_user(
+      max: request.max_consumptions_per_user,
+      is_user_request: !is_nil(balance.user_uuid) && is_nil(balance.account_uuid),
```

Comment:
Can we convert this to a new Balance function: `Balance.is_user_balance?(balance)`?

I think this might be used somewhere else in the future and the logic to check this might change.